PlatformRequirements.md

```diff
@@ -0,0 +1,144 @@
+        - The `build` method must return a promise either fulfilled with an array of build artifacts (application packages) if the package was built successfully, or rejected with a CordovaError. The return value in most cases will contain only one items, but in some cases there could be multiple items in an output array, e.g. when multiple architectures are specified. The resultant build artifact objects are not strictly typed and may contain an arbitrary set of fields as in the sample below.
+ 
+        ```
+        {
+            architecture: 'x86',
+            buildType: 'debug',
+            path: '/path/to/build',
+            type: 'app'
+ 
+        }
+        ```
+ 
+- has a `run` method
+    - PlatformApi.prototype.run = function(runOptions) {};
+    - The `run` method should build an application package for the current platform and runs it on the specified/default device. If no 'device'/'emulator'/'target' options are specified, then it should try to run the app on a default device if connected, otherwise it should run on the app on the emulator.
+    - runOptions : This is an options object. The structure is the same as for build options.
+    - The `run` method must return a promise either fulfilled if the package was build and ran successfully, or rejected with a CordovaError.
+ 
+ 
+- has an `addPlugin` method
+    - PlatformApi.prototype.addPlugin = function (plugin, installOptions) {};
+    - The `addPlugin` method should install a plugin into a platform. It should handle all the non-www files shipped by plugin (sources, libs, assets, js-files) and accept a PluginInfo instance that represents the plugin that will be installed and an options object. It cannot resolve the dependencies of a plugin.
+    - plugin : This is a PluginInfo instance that should represent the plugin that will be installed. It is expected to accept a plugin spec that should be one of the following:
+        - valid plugin id that can be resolved through either cordova plugin registry or npm: 'org.apache.cordova.globalization', 'cordova-plugin-globalization'
+        - valid npm identifier, that resolves to valid plugin : cordova-plugin-globalization@1.0.0
+        - git url, that points to a repo with a valid plugin: http://github.com/apache/cordova-plugin-globalization.git#r.1.0.0
+        - path to local repo of valid plugin: /my/cordova/repositories/cordova-plugin-globalization
+    - installOptions : This is an options object with the following possible options:
+        - installOptions.link : This is a flag that should specify that plugin sources will be symlinked to app's directory instead of copying (if possible).
+        - installOptions.variables : This is an object that should represent variables that will be used to install a plugin.
```

Comment:
Hmm how about this?
https://cordova.apache.org/docs/en/latest/config_ref/#plugin